src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -101,24 +100,38 @@ public async Task<IEnumerable<Type>> GetExtensionsStartupTypesAsync()
                 if (string.IsNullOrEmpty(extensionsPath))
                 {
                     _logger.ScriptStartUpErrorLoadingExtensionBundle();
-                    return new Type[0];
+                    return Array.Empty<Type>();
                 }
+
                 _logger.ScriptStartUpLoadingExtensionBundle(extensionsPath);
             }
             else
             {
                 extensionsPath = Path.Combine(_rootScriptPath, "bin");
 
-                if (!File.Exists(Path.Combine(extensionsPath, ScriptConstants.ExtensionsMetadataFileName)) &&
-                    File.Exists(Path.Combine(_rootScriptPath, ScriptConstants.ExtensionsMetadataFileName)))
+                if (!File.Exists(Path.Combine(extensionsPath, ScriptConstants.ExtensionsMetadataFileName)))
                 {
-                    // As a fallback, allow extensions.json in the root path.
-                    extensionsPath = _rootScriptPath;
+                    if (File.Exists(Path.Combine(_rootScriptPath, ScriptConstants.ExtensionsMetadataFileName)))
+                    {
+                        // As a fallback, allow extensions.json in the root path.
+                        extensionsPath = _rootScriptPath;
+                    }
+                    else
+                    {
+                        string systemPath = Path.Combine(_rootScriptPath, ScriptConstants.AzureFunctionsSystemDirectoryName);
+                        if (File.Exists(Path.Combine(systemPath, ScriptConstants.AzureFunctionsSystemDirectoryName)))
```

Comment:
are you meaning to create a path with `.azureFunctions` here twice? Wouldn't this check for `{root}/.azureFunctions/.azureFunctions`?

Reply:
🤦🏼 last minute change that ended up with paste issue. Will updated this, as it's meant to be a check for the metadata file.